src/components/NotificationModal/RegexSelectorInput.tsx

```diff
@@ -0,0 +1,134 @@
+import { RegexSelector } from "hooks/useNotificationModal";
+import { Select, Input } from "antd";
+import { v4 as uuid } from "uuid";
+import Icon from "@leafygreen-ui/icon";
+import styled from "@emotion/styled";
+
+const { Option } = Select;
+
+export interface RegexSelectorProps {
+  dataCyPrefix?: number;
+  disabledDropdownOptions: string[];
+  dropdownOptions: RegexSelector[];
+  isVisibleDelete?: boolean;
+  key?: string;
+  onChangeRegexValue: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onChangeSelectedOption: (optionValue: string) => void;
+  onDelete: () => void;
+  regexInputValue: string;
+  selectedOption: string;
+}
+
+export const RegexSelectorInput = ({
+  disabledDropdownOptions,
+  dropdownOptions,
+  onChangeRegexValue,
+  onChangeSelectedOption,
+  onDelete,
+  regexInputValue,
+  selectedOption,
+  dataCyPrefix,
+  isVisibleDelete,
+}: RegexSelectorProps) => {
+  const dropdownId = uuid();
+  const inputId = uuid();
+  return (
+    <Container
+      isVisibleDelete={isVisibleDelete}
+      data-cy={`${dataCyPrefix}-regex-selector-container`}
+    >
+      <FlexRow>
+        <div>
+          <div>
+            <InputLabel htmlFor={dropdownId}>Field name</InputLabel>
+          </div>
+          <StyledSelect
+            id={dropdownId}
+            data-test-id={`${dataCyPrefix}-regex-selector-dropdown`}
+            value={selectedOption}
+            onChange={onChangeSelectedOption}
+          >
+            {(dropdownOptions ?? []).map((s) => (
+              <Option
+                key={s.type}
+                disabled={disabledDropdownOptions.includes(s.type)}
+                value={s.type}
+                data-test-id={`${dataCyPrefix}-${s.type}-option`}
+              >
+                {s.typeLabel}
+              </Option>
+            ))}
+          </StyledSelect>
+        </div>
+        <MatchesRegexLabel>matches regex</MatchesRegexLabel>
+      </FlexRow>
+      <RegexContainer>
+        <div>
+          <InputLabel htmlFor={inputId}>Regex</InputLabel>
+        </div>
+        <FlexRow>
+          <StyledInput
+            data-cy={`${dataCyPrefix}-regex-selector-input`}
+            id={inputId}
+            onChange={onChangeRegexValue}
+            value={regexInputValue}
+            disabled={!selectedOption}
+          />
+          <TrashContainer
+            data-cy={`${dataCyPrefix}-regex-selector-trash-container`}
+            isVisibleDelete={isVisibleDelete}
+          >
+            <StyledIcon
+              data-cy={`${dataCyPrefix}-regex-selector-trash`}
+              onClick={onDelete}
+              glyph="Trash"
+            />
+          </TrashContainer>
+        </FlexRow>
+      </RegexContainer>
+    </Container>
+  );
+};
+interface DeleteProps {
+  isVisibleDelete: boolean;
+}
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: row;
+  width: ${(props: DeleteProps): string =>
+    props.isVisibleDelete ? "80%" : "calc(80% - 50px)"};
```

Comment:
the purpose of this code is to align the ends of the regex inputs for all regex selector rows. when the trash can is visible, the width of the row is 80%, otherwise it is 80% minus the width of the trashcan. 
<img width="621" alt="Screen Shot 2020-07-17 at 3 29 30 PM" src="https://user-images.githubusercontent.com/10734386/87825206-acc2cb80-c844-11ea-9bee-02310dc5e46d.png">
<img width="619" alt="Screen Shot 2020-07-17 at 3 29 49 PM" src="https://user-images.githubusercontent.com/10734386/87825209-ad5b6200-c844-11ea-84bb-8b1dc7cbf90e.png">

Reply:
would adding a comment be useful here?